pkg/secrets/aws.go

```diff
@@ -0,0 +1,48 @@
+package secrets
+
+import (
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/pkg/errors"
+	v1 "k8s.io/api/core/v1"
+)
+
+const (
+	// AWSSecretType represent the secret type for AWS credentials.
+	AWSSecretType string = "secrets.kanister.io/aws"
+
+	// AWSAccessKeyID is the key for AWS access key ID.
+	AWSAccessKeyID string = "AWS_ACCESS_KEY_ID"
+	// AWSSecretAccessKey is the key for AWS secret access key.
+	AWSSecretAccessKey string = "AWS_SECRET_ACCESS_KEY"
+	// AWSSessionToken is the key for optional AWS session token.
+	AWSSessionToken string = "AWS_SESSION_TOKEN"
+)
+
+// ExtractAWSCredentials extracts AWS credential values from the given secret.
+//
+// Extracted values from the secrets are:
+// - AWS_ACCESS_KEY_ID (required)
+// - AWS_SECRET_ACCESS_KEY (required)
+// - AWS_SESSION_TOKEN (optional)
+//
+// If the type of the secret is not "secret.kanister.io/aws", it returns an error.
+// If the required types are not avaialable in the secrets, it returns an errror.
+func ExtractAWSCredentials(secret *v1.Secret) (*credentials.Value, error) {
```

Comment:
Let's split Extract from Validate.

[minor] we can also validate that there are not extra fields in data.